main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,506 @@
+The general rule is if you make a new object and give it to someone else (and don't 
+immediately forget it yourself), you should give them `harden(obj)` instead of the raw object. 
+
+After calling lockdown, the harden function ensures that every object in the transitive 
+closure over property and prototype access starting with that object has been frozen by 
+`Object.freeze()`. This means that the object can be passed among programs and none of 
+those programs will be able to tamper with the surface of that object graph. They can 
+only read the surface data and call the surface functions.
+
+You have to harden a class before you harden any of its instances; i.e. it takes two separate 
+steps to harden both a class and its instances. Harden a base class before hardening classes 
+that inherit from it. `harden()` does transitive freezing by following the object’s own
+properties (as opposed to properties it inherited), and the objects whose own properties refer 
+to them, and so forth.
+
+When you freeze an object via `harden()`, it ensures any external callers 
+can only interact with it through functions in the object’s API. `harden()` 
+is an enhanced transitive version of `Object.freeze()`, which only locks up an 
+object's own properties.
+
+All objects that are transferred across a trust boundary must have their API
+surface frozen, usually by calling `harden()`. This ensures other objects can 
+only interact with them through their defined method interface. *CapTP*, our 
+communications layer for passing references to distributed objects, enforces 
+this at vat boundaries.
+
+The general rule is that if you make a new object and give it to someone else 
+(and don't immediately forget it yourself), you should give them `harden(obj)` 
+instead of the raw object. This prevents someone from adding/deleting the properties
+or prototypes of that object. Being hardened doesn't preclude an object from having 
+access to mutable state (`harden(new Map())` still behaves like a normal mutable `Map`), 
+but it means their methods stay the same and can't be surprisingly changed by someone else
+
+Defined objects (`mint`, `issuer`, `zcf`, etc.) shouldn't need hardening as their 
+constructors should do that work. It's mainly records, callbacks, and ephemeral 
+objects that need hardening.
+
+You can send a message to a hardened object. If it's a record, you can access 
+its properties and their values.
+
+`harden()` is automatically provided by SES. Any code that will run inside a vat or a 
+contract can use harden as a global, without importing anything.
+
+Tip: If your text editor/IDE complains about `harden()` not being defined or imported, 
+try adding `/* global harden */` to the top of the file.
+
+You use `harden()` like this:
+```js
+const o = {a: 2};
+o.a  = 12;
+console.log(o.a); // 12 because o is still mutable
+harden(o);
+o.a  = 37; // throws a TypeError because o is now hardened
+```
```

Comment:
```suggestion
> Tip: If your text editor/IDE complains about `harden()` not being defined or imported, 
> try adding `/* global harden */` to the top of the file.
>
> You use `harden()` like this:
> ```js
> const o = {a: 2};
> o.a  = 12;
> console.log(o.a); // 12 because o is still mutable
> harden(o);
> o.a  = 37; // throws a TypeError because o is now hardened
> ```
```

See the preview for the effect. There would be no [disclosure button](https://developer.apple.com/design/human-interface-guidelines/macos/buttons/disclosure-controls/).